light_controller/light_controller.py

```diff
@@ -0,0 +1,195 @@
+        self.toy = IR_Toy(comPort)
+        self._buttons = {}
+        self._buttonsFile = buttons
+        self.loadButtons()
+
+    def loadButtons(self):
+        """
+        Load button codes from the file
+        :return: None
+        """
+        if os.path.exists(self._buttonsFile):
+            with open(self._buttonsFile, 'rb') as fp:
+                buttons = json.load(fp)
+                self._buttons = buttons
+
+    def getSettings(self):
+        """
+        Return the IR Toy settings
+        """
+        return self.toy.getSettings()
+
+    def recordButton(self, button, dump=True):
+        """
+        Record a single button press
+
+        :param button: Name of the button
+        :param dump: Whether to save the current button list
+        """
+        self.toy.reset()
+        self.toy.setSamplingMode()
+        print("Press {}".format(button))
+        sys.stdout.flush()
+        irCode = self.toy.receiveSignal()
+        self._buttons[button] = irCode
+        if dump:
+            self.dumpButtons()
+
+    def dumpButtons(self):
+        """
+        Record All The Buttons!
+        Saves the buttons into the configured button file
+        """
+        with open(self._buttonsFile, 'wb') as fp:
+            json.dump(self._buttons, fp)
+
+    def recordButtons(self):
+        """
+        Record button presses from the remote
+        :return: SFA
+        """
+        print("Protocol Version:", self.toy.protocolVersion)
+        for button in sorted(self.buttonNames):
+            self.recordButton(button, dump=False)
+        self.dumpButtons()
+
+    def lightsOn(self):
+        """
+        Turn the lights on
+        :return:
+        """
+        self.toy.transmitCode(self._buttons['on'])
+
+    def lightsOff(self):
+        """
+        Turn the lights off
+        :return:
+        """
+        self.toy.transmitCode(self._buttons['off'])
+
+    def setColour(self, colour):
+        """
+        Change the light colours
+        :param colour: Colour name from :mod:`getColours`
+        :return: None
+        """
+        if colour in self._colours:
+            self.toy.transmitCode(self._buttons[colour])
+
+    def sendCommand(self, command):
+        """
+        Send a command
+        :param command: Command to send to the lights
+        :return: None
+        """
+        if command in self._commands:
+            self.toy.transmitCode(self._buttons[command])
+
+    def getVersion(self):
+        """
+        Get version information from the IR Toy
+
+        :returns: Hardware version, firmware version
+        """
+        return self.toy.getVersion()
+
+    @property
+    def colours(self):
+        return self._colours
+
+    @property
+    def commands(self):
+        return self._commands
+
+    @property
+    def protocolVersion(self):
+        return self.toy.protocolVersion
+
+
+if __name__ == '__main__':
+    changer = LightController('/dev/ttyACM0')
```

Comment:
Could you abstract this main method into another executable file, or put it behind a command line argument?